fastar/core.py

```diff
@@ -230,3 +233,13 @@ def get_aval(x):
 
 def abstractify_lazy(invals):
   return [abstractify(x._aval) if isinstance(x, LazyArray) else x for x in invals]
+
+class Ones(jc.ShapedArray):
```

Comment:
Is it necessary to inherit from ShapedArray here? Surely the only attribute of Ones we should really be using is the shape (i.e. we'd never need to inspect its dtype).

Reply:
I think it's good to have things like `ndim`, `size` built-in.

Reply:
We can easily implement those ourselves here, to me it feels slightly dangerous to inherit from ShapedArray.

Reply:
It's sort of suggesting to the reader that there might be some deep reason that we're inheriting from ShapedArray, which isn't the case.

Reply:
Done.